python/lsst/cp/pipe/cpBfk.py

```diff
@@ -415,7 +430,52 @@ def run(self, inputPtc, dummy, camera, inputDims):
             if self.config.level == 'DETECTOR' and ampName not in self.config.ignoreAmpsForAveraging:
                 detectorCorrList.extend(scaledCorrList)
                 detectorFluxes.extend(fluxes)
-            bfk.valid[ampName] = True
+
+            # Check for validity
+            if self.config.doCheckValidity:
+                kernelSize = bfk.ampKernels[ampName].shape[0]
+                kernelCenterValue = bfk.ampKernels[ampName][kernelSize//2, kernelSize//2]
+                xv, yv = np.meshgrid(range(kernelSize), range(kernelSize))
+
+                # Get a mask for a 3px picture frame
+                valid = True
+                if kernelSize >= 7:
+                    edges = (xv < 3)
+                    edges += (xv > kernelSize - 3 - 1)
+                    edges += (yv < 3)
+                    edges += (yv > kernelSize - 3 - 1)
+
+                    kernelEdges = bfk.ampKernels[ampName][edges].ravel()
+                    kernelEdgeStd = np.std(kernelEdges)
+
+                    # Edges should converge to zero
+                    valid = np.all(np.isclose(
+                        kernelEdges, 0,
+                        rtol=self.config.nSigmaTolForValidEdge*kernelEdgeStd,
+                    ))
+                    if not valid:
+                        self.log.info(f"{ampName}: the kernel edges did not converge to 0 within "
+                                      f"+/- {self.config.nSigmaTolForValidEdge*kernelEdgeStd} sigma")
```

Comment:
It is against the dev guide and python best practices to use f-strings with logging (I know, very annoying).
You have to do `self.log.info("%s: the kernel...", ampName, ...)`

Also, why is this not a warning?